File: output/llama3.1-8b/avatar/Java/Python/codeforces_7_B.py
import sys

class Memory:
    def __init__(self, m):
        self.memory = [0] * m
        self.alloc_idx = 0

    def alloc(self, n):
        can_alloc = False
        len_ = 0
        for i in range(len(self.memory)):
            if self.memory[i] == 0:
                len_ += 1
            else:
                len_ = 0
            if len_ == n:
                can_alloc = True
                len_ = i - n + 1
                break
        if can_alloc:
            for i in range(len_, len_ + n):
                self.memory[i] = self.alloc_idx
            print(self.alloc_idx)
            self.alloc_idx += 1
        else:
            print("NULL")

    def erase(self, x):
        if x <= 0:
            print("ILLEGAL_ERASE_ARGUMENT")
            return
        has_erased = False
        for i in range(len(self.memory)):
            if self.memory[i] == x:
                self.memory[i] = 0
                has_erased = True
        if not has_erased:
            print("ILLEGAL_ERASE_ARGUMENT")

    def defragment(self):
        d = 0
        for i in range(len(self.memory)):
            if self.memory[i] == 0:
                d += 1
            else:
                self.memory[i - d] = self.memory[i]
        for i in range(len(self.memory) - d, len(self.memory)):
            self.memory[i] = 0

def main():
    t = int(input())
    m = int(input())
    memory = Memory(m)
    for _ in range(t):
        op = input()
        if op == "alloc":
            n = int(input())
            memory.alloc(n)
        elif op == "erase":
            x = int(input())
            memory.erase(x)
        elif op == "defragment":
            memory.defragment()
        else:
            print("h")

if __name__ == "__main__":
    main()

File: output/llama3.1-8b/avatar/Python/Java/codeforces_7_B.java
import java.util.Scanner;
import java.util.Arrays;

public class codeforces_7_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int t = Integer.parseInt(input[0]);
        int m = Integer.parseInt(input[1]);
        int[] alocuente = new int[m];
        Arrays.fill(alocuente, 0);
        int ind = 1;
        for (int i = 0; i < t; i++) {
            String[] op = scanner.nextLine().split(" ");
            if (op[0].equals("alloc")) {
                int pos = 0;
                boolean found = false;
                for (int j = 0; j < m; j++) {
                    if (alocuente[j] == 0) {
                        pos++;
                        if (pos == Integer.parseInt(op[1])) {
                            Arrays.fill(alocuente, j - Integer.parseInt(op[1]) + 1, j + 1, ind);
                            System.out.println(ind);
                            ind++;
                            found = true;
                            break;
                        }
                    } else {
                        pos = 0;
                    }
                }
                if (!found) {
                    System.out.println("NULL");
                }
            }
            if (op[0].equals("erase")) {
                int pos = Integer.parseInt(op[1]);
                if ((pos != 0 && Arrays.binarySearch(alocuente, pos) < 0) || pos == 0) {
                    System.out.println("ILLEGAL_ERASE_ARGUMENT");
                } else {
                    for (int j = 0; j < m; j++) {
                        if (alocuente[j] == pos) {
                            alocuente[j] = 0;
                        }
                    }
                }
            }
            if (op[0].equals("defragment")) {
                int cnt = 0;
                for (int j : alocuente) {
                    if (j == 0) {
                        cnt++;
                    }
                }
                int[] newAlocuente = new int[m - cnt];
                int k = 0;
                for (int j : alocuente) {
                    if (j != 0) {
                        newAlocuente[k++] = j;
                    }
                }
                System.arraycopy(newAlocuente, 0, alocuente, 0, m - cnt);
                Arrays.fill(alocuente, m - cnt, m, 0);
            }
        }
    }
}